### ISSUE TITLE
@struct not working with ES6 classes and goog.module

### ISSUE THREAD

**Issue by supersteves** [google/closure-compiler#1062]
"A.js":
``` javascript
goog.module("A");

/** @struct */
class A {
    /** @struct */
    constructor() {
    }
}

exports = A;
```
(I didn't know where to put ```@struct```, presumably the first place).

"B.js":
``` javascript
goog.module("B");

let A = goog.require("A");

class B extends A {
    constructor() {
    }
}
var b = new B();
b.someWrongProperty = 5;

exports = B;
```

It passes full advanced compilation without failure, but should complain about someWrongProperty on a struct. 

This worked before I switched over from prototype + provide to class + module based definitions.

**Comment by MatrixFrog** [comment]
You can actually omit `@struct`. ES6 classes are treated as structs by default.

My guess is this is a similar cause to https://github.com/google/closure-compiler/issues/1053 -- the alias that's created by the goog.module processing is not quite treated as exactly the same type as the original type it's an alias of.

**Comment by supersteves** [comment]
Ah yes thanks MatrixFrog, I remember reading that. I'll remove ```@struct``` and watch #1053.
